File: backend/src/timeflow/infrastructure/websocket/messages/envelope.py
"""所有 `*.error` 消息共用的错误详情结构。"""

from typing import Any

from pydantic import BaseModel


class ErrorDetail(BaseModel):
    """错误码、说明和上下文,所有 `*.error` 消息的 `error` 字段共用这个形状。"""

    code: str
    message: str
    details: dict[str, Any] | None = None

File: frontend/src/contracts/envelope.ts
export type ApiError = {
  code: string;
  message: string;
  details: Record<string, unknown> | null;
};

export type WsRequest<TType extends string, TPayload> = {
  type: TType;
  request_id: string;
  payload: TPayload;
};

export type WsSuccess<TType extends string, TPayload> = {
  type: TType;
  request_id: string;
  ok: true;
  payload: TPayload;
};

export type WsFailure<TType extends string> = {
  type: TType;
  request_id: string;
  ok: false;
  error: ApiError;
};
